server/src/handlers/bookmarks.rs

```diff
@@ -28,19 +28,32 @@ async fn create_bookmark(
         .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)
 }
 
+async fn delete_bookmark(
+    user: User,
+    State(db): State<DB>,
+    Json(bookmark): Json<Bookmark>,
+) -> Result<Json<Bookmark>, StatusCode> {
+    match db.delete_bookmark(&user.id, bookmark) {
+        Ok(res) => Ok(Json(res)),
+        Err(DbError::NotFound) => Err(StatusCode::NOT_FOUND),
+        Err(DbError::WhoopsieDoopsie) => Err(StatusCode::INTERNAL_SERVER_ERROR),
+    }
+}
+
 pub fn routes() -> Router<DB> {
     Router::<DB>::new()
         .route("/", get(get_bookmarks))
         .route("/", post(create_bookmark))
+        .route("/", delete(delete_bookmark))
```

Comment:
The ID of the bookmark that's being deleted should be part of the path.

Dynamic path arguments can be used elegantly with `axum`, our web framework. Check out the list of [common extractor examples](https://docs.rs/axum/latest/axum/extract/index.html#common-extractors), it has one example for extracting a dynamic `user_id` from the path.